### DISCUSSION TITLE
MATLAB Scripts

### DISCUSSION THREAD

**Discussion by RachelETHOMAS** [simnibs/simnibs#161] [2. General]
Hi there,

A couple more questions regarding scripting SimNIBS in MATLAB,

- Firstly, regarding the electrode size, positioning and orientation - how does the script differentiate the x and y axis from the e.g., s.poslist{2}.electrode(1).dimensions = [30 30]; does it simply read the first value as the x and the second as y? And then how is the s.poslist{2}.electrode(1).pos_ydir = 'Cz';  variable linked to this, to create the orientation (relative to x and y)?

- Secondly, we are keen to know the most efficient way to define individual conductivity values for given tissue layers in the scripting language in matlab. There seems to be a number of possible formats that one could use: however, for our first pass at running SimNIBS via matlab, we would like to leave any uncertainty calcs for now, and just have each tissue layer with a single value. Can you please advise on the proper format within the session struct 's' for this? 
e.g., 
cond(1).value = 0.126;
cond(1).name = 'white matter';
or
e.g., 
% Change White Matter conductivity to 0.1 S/m
tdcs_list.cond(1).value = 0.1;

- Lastly: currently, we are only trying tES simulations before we incorporate both TMS and tES parameters in the same simulatation. However - we are not sure why our tES simulation script needs TMS parameters as well to run. We have tried removing these from the script and this caused errors - can you please advise on the best way forward here? (Please see the attched .zip file with scripts with and without TMS)

Thanks!

[For_GitHub.zip](https://github.com/simnibs/simnibs/files/11939826/For_GitHub.zip)

**Comment by RachelETHOMAS** [comment]
Hi there,

Just thought I'd give a quick update - I've made some progress and worked out how to encode the conductivity values. I found that the format (for WM) 's.poslist{2}.cond(1).value = 0.126;' worked well.

I'm still struggling, particularly, with my last qn- trying to remove TMS without causing it to throw up errors. I was wondering if you guys had made any progress with this?

Thanks so much!

**Comment by TorgeW** [comment]
Looks like you are using the index 2 in the poslist when removing tms.
The poslist should start with the index 1 :)
